agents/suprsync/suprsync.py

```diff
@@ -74,21 +76,33 @@ def run(self, session, params=None):
         handler = SupRsyncFileHandler(
             srfm, self.archive_name, self.remote_basedir, ssh_host=self.ssh_host,
             ssh_key=self.ssh_key, cmd_timeout=self.cmd_timeout,
-            copy_timeout=self.copy_timeout
+            copy_timeout=self.copy_timeout, compression=self.compression,
+            bwlimit=self.bwlimit
         )
 
         self.running = True
         session.set_status('running')
 
         while self.running:
+            op = {'start_time': time.time()}
             try:
-                handler.copy_files(max_copy_attempts=self.max_copy_attempts,
-                                   num_files=self.files_per_batch)
+                session.data['activity'] = 'copying'
+                op['files'] = handler.copy_files(max_copy_attempts=self.max_copy_attempts,
+                                                 num_files=self.files_per_batch)
             except subprocess.TimeoutExpired as e:
                 self.log.error("Timeout when copying files! {e}", e=e)
+                op['error'] = 'timed out'
+            except subprocess.CalledProcessError as e:
+                self.log.error("rsync returned non-zero exit code! {e}", e=e)
+                op['error'] = 'nonzero exit'
```

Comment:
I'm fine with this, though I do think that we should also add a grafana feed that we can use to alert when rsync starts failing. Before we could use the OpCode for the `run` operation to set alerts when things broke, but I think this removes the ability to do that.